src/core/temporal/qgsabstracttemporal.h

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
```

Comment:
I'd suggest moving this class out of the /temporal directory, unless you have plans for adding a few more files into that directory.

Otherwise, there's a few missing pieces needed when adding a new subdirectory:

- Update the doc\CmakeLists.txt file to include the new subdirectory in DOXYGEN_INPUT
- Update python\CMakeLists.txt to include the new subdirectory
- Update tests\src\core\CMakeLists.txt to include it
- Don't use `#include "temporal/qgsabstracttemporal.h", and instead ensure that "#include "qgsabstracttemporal.h"` works instead